### PR TITLE
Fix #118 and #120

### LINKED ISSUE #118: [BUG] Email Migration
## Description  
https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/templates/template-email-migration.html

As of Magento 2.3.4, custom email templates are only allowed to use scalar values for variable data. Direct calls to methods are no longer allowed. To be more specific, methods can no longer be called from variables from either the var directive or when used as parameters. For example {{var order.getEmailCustomerNote()}} or {{something myVar=$obj.method()}} will fail to resolve.

## Prerequisites  

**Magento Version:**
 - >=2.3.4

**Module Version:**
 - any that match Magento >= 2.3.4

### PR DESCRIPTION
**Pull request by elisei**
Fixes the issue of the email template template.

### CODE REVIEW

**Comment by elisei** [comment]
@thomas-kl1 

Can i apply the table to the email template?

This will follow the Luma theme pattern

Ps. Thanks for the excellent module!

File: Model/Customer/Notifier/MailSender.php [hunk 1]
```diff
@@ -16,23 +16,34 @@
...
 use Magento\Store\Model\StoreManagerInterface;
 use Opengento\Gdpr\Model\Notifier\AbstractMailSender;
+use Psr\Log\LoggerInterface;
 
 final class MailSender extends AbstractMailSender implements SenderInterface
 {
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
```
Comment (thomas-kl1) [inline Model/Customer/Notifier/MailSender.php:26]: The latest changes uses typed properties of php7.4. Please could you keep them? The composer.json php restriction is set to min php:7.4

File: Model/Customer/Notifier/MailSender.php [hunk 2]
```diff
@@ -16,23 +16,34 @@
...
      * @var View
      */
-    private View $customerViewHelper;
+    private $customerViewHelper;
 
-    private StoreManagerInterface $storeManager;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
```
Comment (thomas-kl1) [inline Model/Customer/Notifier/MailSender.php:36]: The latest changes uses typed properties of php7.4. Please could you keep them? The composer.json php restriction is set to min php:7.4

File: Model/Order/Notifier/MailSender.php [hunk 3]
```diff
@@ -14,17 +14,25 @@
...
 use Magento\Store\Model\StoreManagerInterface;
 use Opengento\Gdpr\Model\Notifier\AbstractMailSender;
+use Psr\Log\LoggerInterface;
 
 final class MailSender extends AbstractMailSender implements SenderInterface
 {
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
```
Comment (thomas-kl1) [inline Model/Order/Notifier/MailSender.php:24]: The latest changes uses typed properties of php7.4. Please could you keep them? The composer.json php restriction is set to min php:7.4

File: Model/Order/Notifier/MailSender.php [hunk 4]
```diff
@@ -39,9 +47,19 @@ public function send(OrderInterface $order): void
...
+            'customer_data' => [
+                'customer_name' => $order->getCustomerName(),
+            ],
         ];
 
-        $this->sendMail($order->getCustomerEmail(), $order->getCustomerName(), $storeId, $vars);
+        try {
+            $this->sendMail($order->getCustomerEmail(), $order->getCustomerName(), $storeId, $vars);
+            $this->logger->debug(__('GDPR Email Success'));
+        } catch (MailException $exc) {
```
Comment (thomas-kl1) [inline Model/Order/Notifier/MailSender.php:60]: If we catch the exceptions then we should update the phpdoc accordingly

**Review by thomas-kl1 — APPROVED** [review]

**Pull request merged**